### PR TITLE
Bug 1820300: Extend oVirt's MachinePool

### PR DESCRIPTION
**Pull request by rgolangh**
Those items are now part of MachinePool
- instance type
- cpu (cores, sockets)
- memory
- os disk size
- vm type

A compute pool snippet from the install-config.yaml may look like that:

```yaml
compute:
- architecture: amd64
  hyperthreading: Enabled
  name: worker
  platform:
    ovirt:
      cpu:
        cores: 8
        sockets: 1
        instanceTypeID:
      memoryMB: 65536
      osDisk:
        sizeGB: 100
      type: high_performance
  replicas: 3
```

Terraform now uses those values, and all the defaults in tf files are
now deleted in favour of the machine-config ones.

### CODE REVIEW

File: pkg/types/ovirt/machinepool.go [hunk 1]
```diff
@@ -3,11 +3,66 @@ package ovirt
 // MachinePool stores the configuration for a machine pool installed
 // on ovirt.
 type MachinePool struct {
+	// InstanceTypeID defines the VM instance type and overrides
+	// the hardware parameters of the created VM, including cpu and memory.
+	// If InstanceTypeID is passed, all memory and cpu variables will be ignored.
+	InstanceTypeID string `json:"instanceTypeID,omitempty"`
```
Comment (patrickdillon) [inline pkg/types/ovirt/machinepool.go:9]: Can you mark these with +optional and kubebuilder default? See this for details https://github.com/openshift/installer/pull/3515/files#diff-ef5ad052319246b6e805b03c0221dcdcR34 and check out the whole PR for motivation.

File: pkg/types/ovirt/machinepool.go [hunk 2]
```diff
@@ -3,11 +3,66 @@ package ovirt
...
+
+	// Cores is the number of cores per socket.
+	// Total CPUs is (Sockets * Cores * Threads)
+	Cores int32 `json:"cores"`
 }
 
-// Set sets the values from `required` to `a`.
-func (l *MachinePool) Set(required *MachinePool) {
-	if required == nil || l == nil {
+// Disk defines a VM disk
```
Comment (patrickdillon) [inline pkg/types/ovirt/machinepool.go:37]: nit: missing period.

File: pkg/types/ovirt/machinepool.go [hunk 3]
```diff
@@ -3,11 +3,66 @@ package ovirt
...
+type Disk struct {
+	// SizeGB size of the bootable disk in GiB.
```
Comment (patrickdillon) [inline pkg/types/ovirt/machinepool.go:39]: nit: missing `is`

File: pkg/types/ovirt/machinepool.go [hunk 4]
```diff
@@ -3,11 +3,66 @@ package ovirt
...
+
+	// MemoryMB is the size of a VM's memory in MiBs.
+	MemoryMB int32 `json:"memoryMB,omitempty"`
+
+	// OSDisk is the the root disk of the node.
+	OSDisk *Disk `json:"osDisk,omitempty"`
+
+	// VMType defines the workload type the instance will
+	// be used for and this effects the instance parameters.
+	// One of "desktop, server, high_performance"
```
Comment (patrickdillon) [inline pkg/types/ovirt/machinepool.go:22]: `effects` -> `affects`

`One of "desktop, server, high_performance"` -> `One of: "desktop", "server", or "high_performance".`

File: pkg/types/ovirt/machinepool.go [hunk 5]
```diff
@@ -3,11 +3,66 @@ package ovirt
...
+	VMType string `json:"type,omitempty"`
+}
+
+// CPU defines the VM cpu, made of (Sockets * Cores * Threads)
```
Comment (patrickdillon) [inline pkg/types/ovirt/machinepool.go:26]: nit: missing period

File: pkg/types/ovirt/machinepool.go [hunk 6]
```diff
@@ -3,11 +3,66 @@ package ovirt
...
+type CPU struct {
+	// Sockets is the number of sockets for a VM.
+	// Total CPUs is (Sockets * Cores * Threads)
```
Comment (patrickdillon) [inline pkg/types/ovirt/machinepool.go:29]: nit: missing period

File: pkg/asset/machines/master.go [hunk 7]
```diff
@@ -319,7 +319,11 @@ func (m *Master) Generate(dependencies asset.Parents) error {
 		}
 	case ovirttypes.Name:
 		mpool := defaultOvirtMachinePoolPlatform()
+		mpool.VMType = "high_performance"
```
Comment (abhinavdahiya) [inline pkg/asset/machines/master.go:322]: nit: move this to defaultOvirtMachinePoolPlatform

Comment (abhinavdahiya) [reply pkg/asset/machines/master.go:322]: or document why this is different from the default.

File: pkg/types/ovirt/machinepool.go [hunk 8]
```diff
@@ -3,11 +3,66 @@ package ovirt
...
+	// MemoryMB is the size of a VM's memory in MiBs.
+	MemoryMB int32 `json:"memoryMB,omitempty"`
+
+	// OSDisk is the the root disk of the node.
+	OSDisk *Disk `json:"osDisk,omitempty"`
+
+	// VMType defines the workload type the instance will
+	// be used for and this effects the instance parameters.
+	// One of "desktop, server, high_performance"
+	VMType string `json:"type,omitempty"`
```
Comment (abhinavdahiya) [inline pkg/types/ovirt/machinepool.go:23]: if these values are known, `desktop, server, high_performance`, we should be using string enums like the publish https://github.com/openshift/installer/blob/1a0bf0af2eed903d297c2221bea6f256c699004d/pkg/types/installconfig.go#L47-L55

Comment (abhinavdahiya) [reply pkg/types/ovirt/machinepool.go:23]: also include validations.

File: data/data/ovirt/variables-ovirt.tf [hunk 9]
```diff
@@ -46,27 +46,26 @@ variable "ovirt_network_name" {
   description = "The name of ovirt's logical network for the selected ovirt cluster."
 }
 
-variable "ovirt_master_mem" {
-  type    = string
-  default = "8192"
+variable "ovirt_master_memory" {
```
Comment (abhinavdahiya) [inline data/data/ovirt/variables-ovirt.tf:49]: please include descriptions for these global variables.

File: data/data/ovirt/masters/main.tf [hunk 10]
```diff
@@ -1,15 +1,26 @@
...
 
   initialization {
     host_name     = "${var.cluster_id}-master-${count.index}"
     custom_script = var.ignition_master
   }
+
+  block_device {
+    interface = "virtio_scsi"
+    size      = var.ovirt_master_os_disk_size_gb
+  }
```
Comment (abhinavdahiya) [inline data/data/ovirt/masters/main.tf:23]: can you include why this change is part of this PR? in the commit message

File: data/data/ovirt/template/main.tf [hunk 11]
```diff
@@ -51,8 +51,9 @@ resource "ovirt_vm" "tmp_import_vm" {
   count      = length(local.existing_id) == 0 ? 1 : 0
   name       = "tmpvm-for-${ovirt_image_transfer.releaseimage.0.alias}"
   cluster_id = var.ovirt_cluster_id
-  cores      = var.ovirt_template_cpu
-  memory     = var.ovirt_template_mem
+  os {
+    type = "rhcos_x64"
+  }
```
Comment (abhinavdahiya) [inline data/data/ovirt/template/main.tf:56]: https://github.com/openshift/installer/pull/3399/commits/686b239579ca9bfd056db5006168bc58562d43a4#r418711291

File: pkg/types/ovirt/machinepool.go (hunk 1 above)
Comment (abhinavdahiya) [reply pkg/types/ovirt/machinepool.go:9]: > If InstanceTypeID is passed, all memory and cpu variables will be ignored.
I think we need validation that user doesn't set these conflicting values in install-config.yaml

File: data/data/ovirt/masters/main.tf (hunk 10 above)
Comment (rgolangh) [reply data/data/ovirt/masters/main.tf:23]: done

File: data/data/ovirt/template/variables.tf [hunk 12]
```diff
@@ -33,16 +33,3 @@ variable "ovirt_network_name" {
...
-}
-
-variable "ovirt_template_cpu" {
-  type = string
-}
-
-variable "disk_size_gib" {
-  type        = number
-  description = "The size of the template disk for worker/nodes in GiB."
-}
```
[the remark this answers is not in the document]
Comment (rgolangh) [reply data/data/ovirt/template/variables.tf:48]: done

File: data/data/ovirt/masters/main.tf (hunk 10 above)
Comment (rgolangh) [reply data/data/ovirt/masters/main.tf:23]: this will extend the disk size if the machine spec definition is bigger than the definition in the VM template.
will add  to the commit in the commit message

File: data/data/ovirt/masters/variables.tf [hunk 13]
```diff
@@ -22,15 +22,33 @@ variable "ovirt_template_id" {
...
+}
+
+variable "ovirt_master_sockets" {
   type = string
 }
 
-variable "ovirt_master_cpu" {
+variable "ovirt_master_os_disk_size_gb" {
   type = string
 }
```
[the remark this answers is not in the document]
Comment (rgolangh) [reply data/data/ovirt/masters/variables.tf:54]: done

File: data/data/ovirt/variables-ovirt.tf (hunk 9 above)
Comment (rgolangh) [reply data/data/ovirt/variables-ovirt.tf:49]: done

File: data/data/ovirt/variables-ovirt.tf [hunk 14]
```diff
@@ -46,27 +46,26 @@ variable "ovirt_network_name" {
...
-variable "ovirt_template_disk_size_gib" {
-  type    = number
-  default = 25
+variable "ovirt_master_vm_type" {
+  type = string
 }
+
+variable "ovirt_master_instance_type_id" {
+  type = string
+}
```
[the remark this answers is not in the document]
Comment (rgolangh) [reply data/data/ovirt/variables-ovirt.tf:71]: done

File: go.mod [hunk 15]
```diff
@@ -69,12 +69,12 @@ require (
 	github.com/openshift/cluster-api v0.0.0-20191129101638-b09907ac6668
 	github.com/openshift/cluster-api-provider-gcp v0.0.1-0.20200120152131-1b09fd9e7156
 	github.com/openshift/cluster-api-provider-libvirt v0.2.1-0.20191219173431-2336783d4603
-	github.com/openshift/cluster-api-provider-ovirt v0.1.1-0.20200128081049-840376ca5c09
+	github.com/openshift/cluster-api-provider-ovirt v0.1.1-0.20200402173228-4981fa21d0d7
```
[the remark this answers is not in the document]
Comment (rgolangh) [reply go.mod:72]: done

File: pkg/types/ovirt/machinepool.go (hunk 8 above)
Comment (rgolangh) [reply pkg/types/ovirt/machinepool.go:23]: done

File: pkg/types/ovirt/validation/machinepool.go [hunk 16]
```diff
@@ -1,12 +1,48 @@
...
+		}
+	}
+
+	if p.VMType != "" && !p.VMType.ValidValue() {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("vmType"), p.VMType, fmt.Sprintf("VM type must be one of %s", p.VMType.Values())))
+	}
+
+	if p.InstanceTypeID != "" {
+		if p.CPU != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("instanceTypeID"), p.InstanceTypeID, "Mixing instanceTypeID and CPU is not supported"))
```
Comment (patrickdillon) [inline pkg/types/ovirt/validation/machinepool.go:31]: nit: error messages should start with a lowercase letter

File: pkg/types/ovirt/validation/machinepool.go [hunk 17]
```diff
@@ -1,12 +1,48 @@
...
+		}
+		if p.MemoryMB > 0 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("instanceTypeID"), p.InstanceTypeID, "Mixing instanceTypeID and Memory  is not supported"))
```
Comment (patrickdillon) [inline pkg/types/ovirt/validation/machinepool.go:34]: nit: error messages should start with a lowercase letter

File: pkg/types/ovirt/validation/machinepool.go [hunk 18]
```diff
@@ -1,12 +1,48 @@
...
+		}
+		if err := validate.UUID(p.InstanceTypeID); err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("instanceTypeID"), p.InstanceTypeID, err.Error()))
+		}
+	}
+
+	if p.OSDisk != nil {
+		if p.OSDisk.SizeGB <= 0 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("sizeGB"), p.OSDisk.SizeGB, "Disk size must be positive"))
```
Comment (patrickdillon) [inline pkg/types/ovirt/validation/machinepool.go:43]: nit: error messages should start with a lowercase letter

**Comment by patrickdillon** [comment]
@abhinavdahiya PTAL. Needs approval.

File: pkg/types/ovirt/validation/machinepool.go (hunk 16 above)
Comment (rgolangh) [reply pkg/types/ovirt/validation/machinepool.go]: done

File: pkg/types/ovirt/validation/machinepool.go (hunk 17 above)
Comment (rgolangh) [reply pkg/types/ovirt/validation/machinepool.go]: done

File: pkg/types/ovirt/validation/machinepool.go (hunk 18 above)
Comment (rgolangh) [reply pkg/types/ovirt/validation/machinepool.go]: done

File: pkg/asset/installconfig/ovirt/validaton.go [hunk 19]
```diff
@@ -40,9 +40,46 @@ func Validate(ic *types.InstallConfig) error {
 			field.Invalid(ovirtPlatformPath.Child("vnicProfileID"), ic.Ovirt.VNICProfileID, err.Error()))
 	}
 
+	if ic.ControlPlane != nil && ic.ControlPlane.Platform.Ovirt != nil {
+		allErrs = append(
+			allErrs,
+			validateMachinePool(
+				con,
+				field.NewPath("controlPlane", "platform", "ovirt"),
+				ic.ControlPlane.Platform.Ovirt)...)
```
Comment (abhinavdahiya) [inline pkg/asset/installconfig/ovirt/validaton.go:49]: hmm.. can single line this?

File: pkg/asset/installconfig/ovirt/validaton.go [hunk 20]
```diff
@@ -40,9 +40,46 @@ func Validate(ic *types.InstallConfig) error {
...
+	}
+	for idx, compute := range ic.Compute {
+		fldPath := field.NewPath("compute").Index(idx)
+		if compute.Platform.Ovirt != nil {
+			allErrs = append(
+				allErrs,
+				validateMachinePool(
+					con,
+					fldPath.Child("platform", "ovirt"),
+					compute.Platform.Ovirt)...)
```
Comment (abhinavdahiya) [inline pkg/asset/installconfig/ovirt/validaton.go:59]: same as https://github.com/openshift/installer/pull/3399/commits/e002b1a403f43ce70fc6cf2a035fe16ae64f8663#r424176522

File: pkg/types/ovirt/machinepool.go [hunk 21]
```diff
@@ -1,13 +1,99 @@
 package ovirt
 
+import ovirtsdk "github.com/ovirt/go-ovirt"
```
Comment (abhinavdahiya) [inline pkg/types/ovirt/machinepool.go:3]: no SDK dependencies in pkg/types please

File: pkg/types/ovirt/machinepool.go [hunk 22]
```diff
@@ -1,13 +1,99 @@
...
+type Disk struct {
+	// SizeGB size of the bootable disk in GiB.
+	SizeGB int64 `json:"sizeGB"`
 }
 
-// Set sets the values from `required` to `a`.
-func (l *MachinePool) Set(required *MachinePool) {
-	if required == nil || l == nil {
+// VMType defines the type of the VM.
+type VMType string
```
Comment (abhinavdahiya) [inline pkg/types/ovirt/machinepool.go:44]: https://github.com/openshift/installer/blob/master/docs/dev/explain.md#additional-guidelines-on-godoc-comments

File: pkg/types/ovirt/machinepool.go [hunk 23]
```diff
@@ -1,13 +1,99 @@
...
+
+// Values returns a slice of all supported VMTypes.
+func (t VMType) Values() []VMType {
+	return []VMType{
+		VMTypeDesktop,
+		VMTypeServer,
+		VMTypeHighPerformance}
```
Comment (abhinavdahiya) [inline pkg/types/ovirt/machinepool.go:51]:
```suggestion
	return []VMType{
		VMTypeDesktop,
		VMTypeServer,
		VMTypeHighPerformance,
	}
```

File: pkg/types/ovirt/machinepool.go [hunk 24]
```diff
@@ -1,13 +1,99 @@
...
 }
 
-// Set sets the values from `required` to `a`.
-func (l *MachinePool) Set(required *MachinePool) {
-	if required == nil || l == nil {
+// VMType defines the type of the VM.
+type VMType string
+
+// Values returns a slice of all supported VMTypes.
+func (t VMType) Values() []VMType {
```
Comment (abhinavdahiya) [inline pkg/types/ovirt/machinepool.go:47]: Values is not a great name for a function that returns all the valid values.

File: pkg/types/ovirt/machinepool.go [hunk 25]
```diff
@@ -1,13 +1,99 @@
...
+
+// ValidValue returns true if the value is on of the values returned by VMType.Values().
+func (t VMType) ValidValue() bool {
+	for _, v := range t.Values() {
+		if t == v {
+			return true
+		}
+	}
+	return false
+}
```
Comment (abhinavdahiya) [inline pkg/types/ovirt/machinepool.go:62]: validation checking functions shouldn't be part of the API object, please move them to validation package.

File: pkg/types/ovirt/machinepool.go [hunk 26]
```diff
@@ -1,13 +1,99 @@
...
+
+const (
+	// VMTypeDesktop set the VM type to desktop.
+	VMTypeDesktop VMType = VMType(ovirtsdk.VMTYPE_DESKTOP)
```
Comment (abhinavdahiya) [inline pkg/types/ovirt/machinepool.go:66]: https://github.com/openshift/installer/blob/master/docs/dev/explain.md#additional-guidelines-on-godoc-comments (1)

please do not use external dependencies for fields in the InstallConfig.

File: pkg/types/ovirt/validation/machinepool.go [hunk 27]
```diff
@@ -1,12 +1,48 @@
...
+		}
+		if p.CPU.Sockets <= 0 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("sockets"), p.CPU.Sockets, "CPU sockets must be positive"))
+		}
+	}
+
+	if p.VMType != "" && !p.VMType.ValidValue() {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("vmType"), p.VMType, fmt.Sprintf("VM type must be one of %s", p.VMType.Values())))
+	}
+
```
Comment (abhinavdahiya) [inline pkg/types/ovirt/validation/machinepool.go:28]: there is not validation that ensures that the VMType and CPU are not set simultaneously??

Comment (abhinavdahiya) [reply pkg/types/ovirt/validation/machinepool.go:28]: nvm, i see it is here https://github.com/openshift/installer/pull/3399/commits/e002b1a403f43ce70fc6cf2a035fe16ae64f8663#diff-bcbb520f02f11439425a1a60d9d5df3cR30-R32

File: pkg/types/ovirt/validation/machinepool_test.go [hunk 28]
```diff
@@ -20,6 +20,54 @@ func TestValidateMachinePool(t *testing.T) {
...
+			pool: &ovirt.MachinePool{
+				OSDisk: &ovirt.Disk{
+					SizeGB: 0,
+				}},
+			valid: false,
+		},
+		{
+			name: "invalid vmType",
+			pool: &ovirt.MachinePool{
+				VMType: "strong"},
```
Comment (abhinavdahiya) [inline pkg/types/ovirt/validation/machinepool_test.go:52]:
```suggestion
				VMType: "strong"
			},
```

File: pkg/types/ovirt/validation/machinepool_test.go [hunk 29]
```diff
@@ -20,6 +20,54 @@ func TestValidateMachinePool(t *testing.T) {
 			pool:  &ovirt.MachinePool{},
 			valid: true,
 		},
+		{
+			name: "invalid CPU cores",
+			pool: &ovirt.MachinePool{
+				CPU: &ovirt.CPU{
+					Cores:   0,
+					Sockets: 1,
+				}},
```
Comment (abhinavdahiya) [inline pkg/types/ovirt/validation/machinepool_test.go:29]:
```suggestion
				},
			},
```

File: pkg/types/ovirt/validation/machinepool_test.go [hunk 30]
```diff
@@ -20,6 +20,54 @@ func TestValidateMachinePool(t *testing.T) {
...
+			valid: false,
+		},
+		{
+			name: "invalid CPU sockets",
+			pool: &ovirt.MachinePool{
+				CPU: &ovirt.CPU{
+					Cores:   1,
+					Sockets: 0,
+				}},
```
Comment (abhinavdahiya) [inline pkg/types/ovirt/validation/machinepool_test.go:38]:
```suggestion
				},
			},
```

File: pkg/types/ovirt/validation/machinepool_test.go [hunk 31]
```diff
@@ -20,6 +20,54 @@ func TestValidateMachinePool(t *testing.T) {
...
+			valid: false,
+		},
+		{
+			name: "invalid OSDisk",
+			pool: &ovirt.MachinePool{
+				OSDisk: &ovirt.Disk{
+					SizeGB: 0,
+				}},
```
Comment (abhinavdahiya) [inline pkg/types/ovirt/validation/machinepool_test.go:46]:
```suggestion
				},
			},
```

File: pkg/types/ovirt/validation/machinepool_test.go [hunk 32]
```diff
@@ -20,6 +20,54 @@ func TestValidateMachinePool(t *testing.T) {
...
+		{
+			name: "invalid vmType",
+			pool: &ovirt.MachinePool{
+				VMType: "strong"},
+			valid: false,
+		},
+		{
+			name: "invalid instance type id",
+			pool: &ovirt.MachinePool{
+				InstanceTypeID: "aaaaa-123"},
```
Comment (abhinavdahiya) [inline pkg/types/ovirt/validation/machinepool_test.go:58]:
```suggestion
				InstanceTypeID: "aaaaa-123",
			},
```

File: pkg/types/ovirt/validation/machinepool_test.go [hunk 33]
```diff
@@ -20,6 +20,54 @@ func TestValidateMachinePool(t *testing.T) {
...
+			valid: false,
+		},
+		{
+			name: "invalid mix of cpu and instance type id",
+			pool: &ovirt.MachinePool{
+				CPU: &ovirt.CPU{
+					Sockets: 1,
+					Cores:   4,
+				},
+				InstanceTypeID: "85c65199-2df1-43bf-94f6-7e1567e6b238"},
```
Comment (abhinavdahiya) [inline pkg/types/ovirt/validation/machinepool_test.go:68]:
```suggestion
				InstanceTypeID: "85c65199-2df1-43bf-94f6-7e1567e6b238",
			},
```

File: data/data/ovirt/template/variables.tf (hunk 12 above)
[the remark this answers is not in the document]
Comment (abhinavdahiya) [reply data/data/ovirt/template/variables.tf:52]: i can still see the newline :P

File: pkg/asset/installconfig/ovirt/validaton.go [hunk 34]
```diff
@@ -40,9 +40,46 @@ func Validate(ic *types.InstallConfig) error {
...
+func validateMachinePool(con *ovirtsdk.Connection, child *field.Path, pool *ovirt.MachinePool) field.ErrorList {
+	allErrs := field.ErrorList{}
+	allErrs = append(allErrs, validateInstanceTypeID(con, child, pool))
+	return allErrs
+}
+
+func validateInstanceTypeID(con *ovirtsdk.Connection, child *field.Path, machinePool *ovirt.MachinePool) *field.Error {
+	if machinePool.InstanceTypeID != "" {
+		_, err := con.SystemService().
+			InstanceTypesService().InstanceTypeService(machinePool.InstanceTypeID).Get().Send()
```
Comment (abhinavdahiya) [inline pkg/asset/installconfig/ovirt/validaton.go:75]: I'm not big fan of mixed indentation here, one multi-line the rest is one-line. single line is better even for this imo.

File: pkg/types/ovirt/machinepool.go (hunk 5 above)
Comment (abhinavdahiya) [reply pkg/types/ovirt/machinepool.go:26]:
```suggestion
// CPU defines the VM's cpu, which is made of (Sockets * Cores * Threads)
```

The sockets and cores are defined in this struct, while there is no mention of the threads?

File: pkg/types/ovirt/validation/machinepool.go [hunk 35]
```diff
@@ -1,12 +1,48 @@
...
+	if p.InstanceTypeID != "" {
+		if p.CPU != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("instanceTypeID"), p.InstanceTypeID, "mixing instanceTypeID and CPU is not supported"))
+		}
+		if p.MemoryMB > 0 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("instanceTypeID"), p.InstanceTypeID, "mixing instanceTypeID and Memory  is not supported"))
```
Comment (abhinavdahiya) [inline pkg/types/ovirt/validation/machinepool.go:34]:
```suggestion
			allErrs = append(allErrs, field.Invalid(fldPath.Child("instanceTypeID"), p.InstanceTypeID, "mixing instanceTypeID and Memory is not supported"))
```

**Comment by abhinavdahiya** [comment]
Please update the godoc comments for the new fields in the machinepool as these are very sparse and incomplete feeling. With the addition of `openshift-install explain installconfig` https://github.com/openshift/installer/pull/3515 the godoc comments will directly relate to better UX for users.

Make sure include the generated code using https://github.com/openshift/installer/blob/master/docs/dev/explain.md#generating-the-documentation

also, there is missing documentation in docs/ovirt/customization.md that documents the fields and also provides some example install-config.yaml 's for various configurations for users. see https://github.com/openshift/installer/blob/master/docs/user/aws/customization.md for AWS as example.

File: pkg/types/ovirt/machinepool.go (hunk 24 above)
Comment (abhinavdahiya) [reply pkg/types/ovirt/machinepool.go:47]: also for consistency sake with other fields in InstallConfig, I would recommend we keep the list in the validation package.

File: pkg/asset/installconfig/ovirt/validaton.go (hunk 19 above)
Comment (rgolangh) [reply pkg/asset/installconfig/ovirt/validaton.go:49]: done.
The reason I do those if for lines longer than 120.  I see no guidelines on the project for this and I think its much easier to read it like that.

File: pkg/types/ovirt/machinepool.go (hunk 21 above)
Comment (rgolangh) [reply pkg/types/ovirt/machinepool.go:3]: done

File: pkg/types/ovirt/machinepool.go (hunk 23 above)
Comment (rgolangh) [reply pkg/types/ovirt/machinepool.go:51]: done

File: pkg/types/ovirt/validation/machinepool_test.go (hunk 28 above)
Comment (rgolangh) [reply pkg/types/ovirt/validation/machinepool_test.go:52]: done

File: pkg/types/ovirt/validation/machinepool_test.go (hunk 30 above)
Comment (rgolangh) [reply pkg/types/ovirt/validation/machinepool_test.go:38]: done

File: pkg/types/ovirt/validation/machinepool_test.go (hunk 29 above)
Comment (rgolangh) [reply pkg/types/ovirt/validation/machinepool_test.go:29]: done

File: pkg/types/ovirt/validation/machinepool_test.go (hunk 31 above)
Comment (rgolangh) [reply pkg/types/ovirt/validation/machinepool_test.go:46]: done

File: pkg/types/ovirt/validation/machinepool_test.go (hunk 33 above)
Comment (rgolangh) [reply pkg/types/ovirt/validation/machinepool_test.go:68]: done

File: pkg/asset/installconfig/ovirt/validaton.go (hunk 34 above)
Comment (rgolangh) [reply pkg/asset/installconfig/ovirt/validaton.go:75]: for me shorter lines (< 120) are preferred for the price of a bit of weirdness of go multi-line. Unfortunately ovirt sdk is more verbose than I'd wanted it to be (a lot of client code with builder probably)

Comment (rgolangh) [reply pkg/asset/installconfig/ovirt/validaton.go:75]: For me shorter lines (< 120) are preferred for the price of a bit of weirdness of go multi-line. Unfortunately ovirt sdk is quite verbose (a lot of client code are the same probably...)

If that's a code-base guideline that's fine by me. I don't want to repeat this over again.

File: pkg/types/ovirt/machinepool.go (hunk 5 above)
Comment (rgolangh) [reply pkg/types/ovirt/machinepool.go:26]: I'm hiding threads for now, I don't want to expose this at the moment. I removed it from the comments as well.

File: pkg/types/ovirt/machinepool.go (hunk 6 above)
Comment (rgolangh) [reply pkg/types/ovirt/machinepool.go:29]: done
